web/views.py

```diff
@@ -38,33 +42,46 @@ def compare(request):
 
     lang1 = request.GET.get('lang1', '')
     lang2 = request.GET.get('lang2', '')
-    lang1_directory = meta_data_langs[lang1]
-    lang2_directory = meta_data_langs[lang2]
-
-    concept = request.GET.get('concept', '')
+    lang1_directory = meta_data_langs.get(lang1)
+    lang2_directory = meta_data_langs.get(lang2)
 
-    with open("web/thesauruses/" + lang1_directory + "/" + concept + ".json", 'r') as lang1_file:
-        data = lang1_file.read()
-    # parse file
-    lang1_file_json = json.loads(data)
-    lang1_concept = lang1_file_json[concept]
-    lang1_friendlyname = lang1_file_json["meta"]["language_name"]
+    if not (lang1_directory and lang2_directory):
+        return HttpResponseBadRequest("Lang does not exist")
 
-    with open("web/thesauruses/" + lang2_directory + "/" + concept + ".json", 'r') as lang2_file:
-        data = lang2_file.read()
-    # parse file
-    lang2_file_json = json.loads(data)
-    lang2_concept = lang2_file_json[concept]
-    lang2_friendlyname = lang2_file_json["meta"]["language_name"]
+    concept = request.GET.get('concept', '')
 
-    common_concepts =[]
-    for key in list(set.union(set(lang1_concept.keys()), set(lang2_concept.keys()))):
+    lang1_file_path = os.path.join(
+        "web", "thesauruses", lang1_directory, concept) + ".json"
+    lang2_file_path = os.path.join(
+        "web", "thesauruses", lang2_directory, concept) + ".json"
+
+    try:
```

Comment:
Thanks for adding this `try` block too. It was another thing I hadn't converted over from the old framework!